libc/include/__llvm-libc-common.h

```diff
@@ -52,6 +52,16 @@
 #undef __restrict
 #define __restrict restrict // C99 and above support the restrict keyword.
 
+#undef _Noreturn
+#if __STDC_VERSION__ >= 201112L
+// In C11 and later, _Noreturn is a keyword.
```

Comment:
woah, so I was familiar with the keyword, but for some reason I had a hunch there was also a function attribute using the C23 `[[]]` syntax...turns out there is! lol

So there's BOTH they keyword / "function specifier" `_Noreturn`, AND the function attribute `[[noreturn]]` (sounds like `[[_Noreturn]]` is supported but "obsolescent." See footnote 175 of n3220) (and the GNU function attribute syntax). Looks like C23 also added `__has_c_attribute` preprocessor directive, but I'm guessing that's not yet widely supported by many versions of gcc+clang.

Reply:
https://fosstodon.org/@llvm/113788196531306328